app/assets/javascripts/discourse/app/lib/click-track.js

```diff
@@ -34,6 +34,35 @@ export function isValidLink($link) {
   );
 }
 
+export function shouldOpenInNewTab(href) {
+  const isInternal = DiscourseURL.isInternal(href);
+  const openExternalInNewTab = User.currentProp("external_links_in_new_tab");
+  return !isInternal && openExternalInNewTab;
+}
+
+export function openLinkInNewTab(link) {
+  let href = (link.href || link.dataset.href || "").trim();
+  const newWindow = window.open(href, "_blank");
+  newWindow.opener = null;
+  newWindow.focus();
+
+  // Hack to prevent changing current window.location.
+  // e.preventDefault() does not work.
+  if (!link.dataset.href) {
+    link.classList.add("no-href");
+    link.setAttribute("data-href", link.href);
```

Comment:
I think the best now is to use dataset https://developer.mozilla.org/fr/docs/Web/API/HTMLElement/dataset